File: AOJ/ITP1/Solved/ITP1_4_D.cpp
#include <cstdio>

int a[15000];

int main()
{
	int N, min = 99999999, max = -99999999;
	long long int sum = 0;

	scanf(" %d", &N);

	for (int i = 0; i < N; ++i)
	{
		scanf(" %d", &a[i]);
	}

	for (int i = 0; i < N; ++i)
	{
		sum += a[i];
		if (max < a[i]) {
			max = a[i];
		}
		if (min > a[i]) {
			min = a[i];
		}
	}

	printf("%d %d %lld\n", min, max, sum);

	return 0;
}

File: AOJ/ITP1/Solved/ITP1_4_D.java
import java.io.*;
import java.util.*;

class Main
{
    public static void main(String[] args)
    {
        Scanner sc = new Scanner(System.in);
        
        int n = sc.nextInt();
        int[] array = new int[n];
        
        for (int i = 0; i < n ; i++) {
            array[i] = sc.nextInt();
        }
        
        int max = array[0];
        int min = array[0];
        long sum = 0;
        
        for (int i = 0; i < n; i++) {
            if (max < array[i]) {
                max = array[i];
            }
            if (min > array[i]) {
                min = array[i];
            }
            sum += array[i];
        }
        
        System.out.printf("%d %d %d\n", min, max, sum);
    }
}